unity-renderer/Assets/UnityGLTF/Scripts/GLTFComponent.cs

```diff
@@ -339,29 +306,36 @@ public IEnumerator LoadAssetCoroutine(Settings settings)
                             sceneImporter?.Dispose();
                             sceneImporter = null;
                         }
-
-                        loader = null;
                     }
 
                     alreadyLoadedAsset = true;
 
-                    CoroutineStarter.Stop(loadingRoutine);
-                    loadingRoutine = null;
-
-                    if ( state == State.COMPLETED )
+                    if ( state == State.COMPLETED && !token.IsCancellationRequested)
                         OnFinishedLoadingAsset?.Invoke();
                     else
                         OnFailedLoadingAsset?.Invoke(new Exception($"GLTF state finished as: {state}"));
 
                     Destroy(loadingPlaceholder);
                     Destroy(this);
+                    CleanUp();
                 }
             }
             else
             {
                 Debug.Log("couldn't load GLTF because url is empty");
             }
         }
+        private void RemoveFromQueue()
```

Comment:
```suggestion
        private void DequeueDownload()
```
When you have two methods that are twins (in this case for Queuing and Dequeuing), is more legible to try to match their names and use opposite verbs that are consistent through the context.